platforms/common/src/kernels/gbsaObc.cc

```diff
@@ -1,12 +1,25 @@
 #define WARPS_PER_GROUP (FORCE_WORK_GROUP_SIZE/TILE_SIZE)
 
+#if defined(USE_HIP)
+
+typedef struct alignas(16) {
```

Comment:
Since the structures are identical other than alignment, what about writing it like this?

```c++
#if defined(USE_HIP)
    #define ALIGN alignas(16)
#else
    #define ALIGN
#endif

typedef struct ALIGN {
```

That way we don't repeat the structure definition, which risks someone changing it in one place but not the other.

Reply:
You are right! Done.

Regarding other structs: I can use the same approach with defines to hide the differences (alignment, paddings) but they also have a different order of fields. I'm worried to rearrange fields for all platforms because I'm not sure that it won't affect performance. I can profile GBSA and Amoeba benchmarks on a few Nvidia GPUs with CUDA: 3090, Titan V, 2060, but definitely this can't cover all devices where OpenMM is used.
In my opinion, it seems safer to keep separate definitions. Also it's likely that we'll be able to remove at least some of this commit in the future because (as far as I know) developers of the HIP compiler continue to work on improving code generation of shared memory accesses.

What do you think?